__tests__/app.test.js

```diff
@@ -241,37 +241,165 @@ describe("APP", () => {
-            expect(body.msg).toBe('bad post request - comment format error')
+            expect(body.msg).toBe("bad post request - comment format error");
           });
       });
-    })
+      describe("QUERIES", () => {
+        test("Query returns and array", () => {
+          return request(app)
+            .get("/api/articles/?filter_by=cats")
+            .expect(200)
+            .then(({ body }) => {
+              expect(Array.isArray(body.articles)).toEqual(true);
+            });
+        });
+        test("Query to return articles by topic", () => {
+          return request(app)
+            .get("/api/articles/?topic=mitch")
+            .expect(200)
+            .then(({ body }) => {
+              body.articles.forEach((article) => {
+                expect(article.topic).toBe("mitch");
+              });
+            });
+        });
+        test("Query to return articles by topic with correct properties", () => {
+          return request(app)
+            .get("/api/articles/?topic=cats")
+            .expect(200)
+            .then(({ body }) => {
+              body.articles.forEach((article) => {
+                expect(article).toEqual({
+                  article_id: expect.any(Number),
+                  title: expect.any(String),
+                  topic: "cats",
+                  author: expect.any(String),
+                  body: expect.any(String),
+                  created_at: expect.any(String),
+                  comment_count: expect.any(String),
+                  article_img_url: expect.any(String),
+                  votes: expect.any(Number),
+                });
+              });
+            });
+        });
+        test("Query to sort by any column", () => {
+          return request(app)
+            .get("/api/articles/?sort=comment_count")
+            .expect(200)
+            .then(({ body }) => {
+              for (let i = 0; i < body.articles.length - 1; i++)
+                expect(+body.articles[i].comment_count).toBeLessThanOrEqual(
+                  +body.articles[i + 1].comment_count
+                );
+            });
+        });
+        test("Query to sort by any column 2", () => {
+          return request(app)
+            .get("/api/articles/?sort=votes")
+            .expect(200)
+            .then(({ body }) => {
+              for (let i = 0; i < body.articles.length - 1; i++)
+                expect(+body.articles[i].votes).toBeLessThanOrEqual(
+                  +body.articles[i + 1].votes
+                );
+            });
+        });
+        test("Query to order by decending", () => {
+          return request(app)
+            .get("/api/articles/?order=asc")
+            .expect(200)
+            .then((({ body }) => {
+              for (let i = 0; i < body.articles.length - 1; i++)
+                expect(new Date(body.articles[i].created_at)).toBeBefore(
+                  new Date(body.articles[i + 1].created_at)
+                );
```

Comment:
nicely done; jest-sorted is neater, probably, but this is some neat problem solving